services/web-app/utils/mdx-image-optimizer-plugin.mjs

```diff
@@ -4,35 +4,69 @@
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
+import path from 'path'
 import { getPlaiceholder } from 'plaiceholder'
+import { remove } from 'unist-util-remove'
 import { visit } from 'unist-util-visit'
 
+const urlsBelongToTheSameRepo = (url1, url2) => {
+  // first four chunks of urls must be equal:
+  // https://github.com/org/repo/*
+  return (
+    JSON.stringify(
+      url1
+        .split('/')
+        .filter((chunk) => !!chunk)
+        .slice(0, 4)
+    ) ===
+    JSON.stringify(
+      url2
+        .split('/')
+        .filter((chunk) => !!chunk)
+        .slice(0, 4)
+    )
+  )
+}
```

Comment:
We should chat through this. Is there an opportunity to use the URL lib here or a regular expression instead?

Reply:
Would this be vulnerable to a URL such as `https://github.com/carbon-design-system/carbon/packages/react/../../../malicious-org/bad-repo/uh-oh.jpg`?

Reply:
sure! I don't see how we could use the url lib but I'm open to try a regex I just don't know how...
 the reason I'm having to do this is because the dirPath looks something like:
`https://github.com/carbon-design-system/carbon-website/blob/main/src/pages/components/accordion` and after I create the imagePath with path.join it looks like this `https:/github.com/carbon-design-system/carbon-website/blob/main/src/pages/components/accordion/images/accordion-style-2.png/?raw=true` so I need to compare the first four chunks of the url and I can't do that directly because for some reason path.join is stripping the second '/' after https

I am using path.join now, so a url like `https://github.com/carbon-design-system/carbon/packages/react/../../../malicious-org/bad-repo/uh-oh.jpg` will be converted to `https://github.com/carbon-design-system/malicious-org/bad-repo/uh-oh.jpg` and that will not pass the check

Reply:
<img width="874" alt="image" src="https://user-images.githubusercontent.com/51208233/169611459-d3c6f3bc-2190-4415-9ce7-af50e4f7987b.png">

Some food for thought for the URL stuff. The good part about the URL class is that you can modify the pieces individually and then output the result via `toString()`. So you can look at the hostname in isolation, make comparions/changes/etc. and do the same thing for the pathname separately without risking collision between the two.